api/src/main/java/com/datastrato/gravitino/authorization/SecurableObjects.java

```diff
@@ -29,36 +34,90 @@ public static SecurableObject of(String... names) {
       throw new IllegalArgumentException("Cannot create a securable object with no names");
     }
 
+    if (type == null) {
+      throw new IllegalArgumentException("Cannot create a securable object with no type");
+    }
+
+    if (names.length > 3) {
+      throw new IllegalArgumentException(
+          "Cannot create a securable object with the name length which is greater than 3");
+    }
+
+    if (names.length == 1
+        && type != SecurableObject.Type.CATALOG
+        && type != SecurableObject.Type.METALAKE) {
+      throw new IllegalArgumentException(
+          "If the length of names is 1, it must be the CATALOG or METALAKE type");
+    }
+
+    if (names.length == 2 && type != SecurableObject.Type.SCHEMA) {
+      throw new IllegalArgumentException("If the length of names is 2, it must be the SCHEMA type");
+    }
+
+    if (names.length == 3
+        && type != SecurableObject.Type.FILESET
+        && type != SecurableObject.Type.TABLE
+        && type != SecurableObject.Type.TOPIC) {
+      throw new IllegalArgumentException(
+          "If the length of names is 3, it must be FILESET, TABLE or TOPIC");
+    }
+
+    List<SecurableObject.Type> types = Lists.newArrayList(type);
+
+    // Find all the types of the parent securable object.
+    SecurableObject.Type curType = type;
+    for (int parentNum = names.length - 2; parentNum >= 0; parentNum--) {
+      curType = getParentSecurableObjectType(curType);
+      types.add(curType);
+    }
```

Comment:
This part of code is a bit of hardcoded, can you please refine this code and remove this hardcoded numbers.

Reply:
Done.